packages/fuels-core/src/abi_encoder.rs

```diff
@@ -65,8 +66,19 @@ impl ABIEncoder {
         self.encode_tokens(arg_array)
     }
 
-    fn encode_string(&mut self, arg_string: &str) {
-        self.buffer.extend(pad_string(arg_string));
+    fn encode_string(&mut self, arg_string: &StringToken) {
+        if !arg_string.0.is_ascii() {
+            panic!("String parameters can only have ascii values");
```

Comment:
Since we're leaving for the user the responsibility of passing appropriately sized strings to the encoder, I'm not sure we should panic in case he does not do so mostly because it doesn't seem to me to be such an unexpected thing to happen, especially since we're still using rust Strings.

A Result might be better.

If we were not building an SDK I'd say leave the panic, because it points to the error of the programmer for not guarding against this, and panicking would be a fail-early approach.

But this is a library, hate to make such decisions for our users -- don't know how important their process is for us to just kill it by panicking.

Reply:
I agree that at this part of the code a Result would be better and I added it. However, the panic is still going to happen because of the way the contract methods are built. The `method_hash` function from `fuels-contract` calls an `unwrap` on the encoder result: `let encoded_args = ABIEncoder::encode(args).unwrap();` and IMO this is ok as we should stop if there are any errors within the encoder.